app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,44 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(benchmark_run_id, benchmark_type_id, benchmark_result_type_id)
+
+    benchmark_run = BenchmarkRun.find_by_id(benchmark_run_id)
+    benchmark_type = BenchmarkType.find_by_id(benchmark_type_id)
+    benchmark_result_type = BenchmarkResultType.find_by_id(benchmark_result_type_id)
+
+    category = benchmark_run.result.keys[0]
+    current_id = benchmark_run_id
+    last_id = BenchmarkRun.where(initiator_type: Commit, benchmark_type: benchmark_type,
+                     benchmark_result_type: benchmark_result_type).order(:created_at).first.id
+
+    previous_benchmark_runs = BenchmarkRun.where("id not in (?)",(current_id..last_id)).where(initiator_type: Commit, benchmark_type: benchmark_type,
```

Comment:
I mentioned this previously. You can't use the id range to fetch the benchmark runs. You need to fetch the benchmark runs of the last 1000 commits ordered by the created at date of the commit.